docs/src/components/BuiltinSearch/index.js

```diff
@@ -0,0 +1,155 @@
+
+          const anchor = `builtin-${category}-${builtin.replaceAll(".", "")}`;
+
+          const isInfix = !!fn.infix;
+          const isRelation = !!fn.relation;
+
+          const args = fn.args || [];
+          const result = fn.result || {};
+
+          const signature = isInfix
+            ? `${args[0]?.name || "x"} ${fn.infix} ${args[1]?.name || "y"}`
+            : isRelation
+            ? `${builtin}(${args.map((a) => a.name).join(", ")}, ${result.name})`
+            : `${result.name || "result"} := ${builtin}(${args.map((a) => a.name).join(", ")})`;
+          return {
+            name: builtin,
+            category: category,
+            wasm: fn.wasm,
+            introduced: fn.introduced,
+            infix: isInfix,
+            signature: signature,
+            anchor: anchor,
+            versions: fn.available,
+            deprecated: fn?.deprecated ?? false,
+            internal: category == "internal",
+          };
+        });
+      })).reduce((a, b) => a.concat(b), []); // builds and flattens the function list
+  });
+  const filteredRows = allRows.map((row) => {
+    return (
+        (row.name.includes(searchTerm) || row.signature.includes(searchTerm) && row.infix) // filter name
+        && (!selectedCategory || row.category == selectedCategory) // filter category
+        && (!isWasm || row.wasm) // filter wasm
+      )
+      ? row
+      : null;
+  }).filter((a) => (
+    a != null
+    && !(a.internal || a.deprecated) // ignore internal and deprecated functions
+    && (searchTerm || selectedCategory || isWasm) // ensures that at least one criteria is being filtered on
+  ));
+  return (
+    <>
+      <div className={styles.searchBar}>
+        <select
+          title="Filter Catgeory"
+          value={selectedCategory}
+          onChange={(e) => setSelectedCategory(e.target.value)}
+        >
+          <option value="" key="0">All Categories</option>
+          {Array.from(Object.keys(builtins._categories).entries()).map((
+            mod,
+          ) => (mod[1] != "internal" && <option key={mod[0] + 1} value={mod[1]}>{mod[1]}</option>))}
+        </select>
+        <input
+          type="text"
+          title="Function Name"
+          placeholder="Search built-ins..."
+          value={searchTerm}
+          onChange={(e) => setSearchTerm(e.target.value)}
+        />
+        <span className={isWasm ? styles.wasm_1 : styles.wasm_0} onClick={(e) => setIsWasm(!isWasm)}>
+          {isWasm ? "✓" : "✗"} Wasm Only
+        </span>
+      </div>
+      {filteredRows.length > 0
+          && (((!entryLimit || filteredRows.length <= entryLimit) && (renderTable(filteredRows)))
+            || ((searchTerm || selectedCategory || isWasm) && filteredRows.length > entryLimit && (
+              <p>Currently {filteredRows.length} matches, keep typing to narrow it down</p>
+            ))) || ((searchTerm || selectedCategory || isWasm) && filteredRows.length == 0 && <p>No matches</p>)}
```

Comment:
something like this might be easier to read?

```js
if (filteredRows.length === 0) {
  if (searchTerm || selectedCategory || isWasm) {
    return <p>No matches</p>;
  }
  return null;
}

if (!entryLimit || filteredRows.length <= entryLimit) {
  return renderTable(filteredRows);
}

if (searchTerm || selectedCategory || isWasm) {
  return <p>Currently {filteredRows.length} matches, keep typing to narrow it down</p>;
}

return null;
```